File: get-simulator-token.py
#!/usr/bin/env python3
"""
Generate a fresh JWT token for the traffic simulator.
Usage: python get-simulator-token.py [email]
"""
import jwt
import time
import sys
import os
from pathlib import Path
from dotenv import load_dotenv

# Load server .env to get JWT_SECRET
load_dotenv(Path(__file__).parent / 'server' / '.env')

JWT_SECRET = os.getenv('JWT_SECRET', 'change_me')
email = sys.argv[1] if len(sys.argv) > 1 else 'simulator@traffic-intel.local'

# Create a token valid for 30 days
payload = {
    'email': email,
    'sub': '999',  # Reserved simulator user ID
    'iat': int(time.time()),
    'exp': int(time.time()) + (30 * 24 * 60 * 60)  # 30 days
}

token = jwt.encode(payload, JWT_SECRET, algorithm='HS256')

print('Generated Simulator JWT Token:')
print('=' * 80)
print(token)
print('=' * 80)
print(f'Valid for: 30 days')
print(f'Email: {email}')
print('\nUpdate traffic-simulator/.env with:')
print(f'SIM_AUTH_TOKEN={token}')

File: get-simulator-token.js
#!/usr/bin/env node
/**
 * Generate a fresh JWT token for the traffic simulator.
 * Usage: node get-simulator-token.js [email]
 * Default email: simulator@traffic-intel.local
 */
const jwt = require('jsonwebtoken');
// Use server config so the same JWT secret is used as the running server
const serverConfig = require('./server/src/config/config');

const JWT_SECRET = (serverConfig && serverConfig.auth && serverConfig.auth.jwtSecret)
  ? serverConfig.auth.jwtSecret
  : (process.env.JWT_SECRET || 'change_me');
const email = process.argv[2] || 'simulator@traffic-intel.local';

// Create a token valid for 30 days
const token = jwt.sign(
  { email, sub: '999' }, // sub=999 is a reserved simulator user ID
  JWT_SECRET,
  { expiresIn: '30d' }
);

console.log('Generated Simulator JWT Token:');
console.log('================================');
console.log(token);
console.log('================================');
console.log(`Valid for: 30 days`);
console.log(`Email: ${email}`);
console.log('\nUpdate traffic-simulator/.env with:');
console.log(`SIM_AUTH_TOKEN=${token}`);
